app.rb

```diff
@@ -0,0 +1,92 @@
+require_relative 'book'
+require_relative 'classroom'
+require_relative 'person'
+require_relative 'rental'
+require_relative 'student'
+require_relative 'teacher'
+class App
+  def initialize
+    @books = []
+    @people = []
+  end
+
+  def create_person(person_code)
+    if person_code == 1
+      create_student
+    elsif person_code == 2
+      create_teacher
+    else
+      puts ' invalid input'
+      call(3)
+    end
+    puts 'Person created successfully'
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    has_permission = gets.chomp
+    permission_values = %w[n N]
+    student = Student.new(age, permission_values.include?(has_permission), name)
+    @people.push(student)
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(age, name, specialization)
+    @people.push(teacher)
+  end
+
+  def create_book(title, author)
+    book = Book.new(title, author)
+    @books.push(book)
+    puts 'Book created successfully'
+  end
+
+  def list_all_books
+    @books.each do |book|
+      puts "Title: \"#{book.title}\", Author: #{book.author}"
+    end
+  end
+
+  def list_all_people
+    @people.each do |person|
+      type = person.instance_of?(Student) ? 'Student' : 'Teacher'
+      puts "[#{type}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number'
+    @books.each_with_index do |book, index|
+      puts "#{index}) Title: \"#{book.title}\", Author: #{book.author}"
+    end
+    book_index = gets.chomp.to_i
+    puts 'Select a person from the following list by number (not id)'
+    @people.each_with_index do |person, index|
+      type = person.instance_of?(Student) ? 'Student' : 'Teacher'
+      puts "#{index}) [#{type}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    person_index = gets.chomp.to_i
+    print 'Date: '
+    date = gets.chomp
+    Rental.new(date, @books[book_index], @people[person_index])
+    puts 'Rental created successfully'
+  end
+
+  def get_rental(id)
+    person = people.find { |person_id| person_id.id == id }
+    puts 'Rentals:'
```

Comment:
- [ ] I received an error while attempting to list rentals by a person ID. To resolve this, I recommend updating the code from `person = people.find { |person_id| person_id.id == id }` to `person = @people.find { |person_id| person_id.id == id }`. This change should resolve the issue. Thank you.

  
  ![Screenshot from 2023-08-30 22-18-19](https://github.com/manzitresor/school_library/assets/92303891/e4afb9a2-b232-4876-a502-0b55f002242f)